### DISCUSSION TITLE
how to send a break signale via asyncio.StreamWriter function

### DISCUSSION THREAD

**Discussion by skylin008** [micropython/micropython#12774] [STM32 / Pyboard]
Thanks [peterhinch](https://github.com/peterhinch) to shared the uasyncio impressive project . I had write a LIN bus driver used the asyncio.StreamWriter, the LIN driver has a break signal, how to send the break signal via asyncio.StreamWriter. thanks!

**Comment by peterhinch** [comment]
I don't think you can do this. As you probably know, `break` is not a character but a sustained logic 0 on the interface. So you'll have to use [uart.sendbreak](https://docs.micropython.org/en/latest/library/machine.UART.html#machine.UART.sendbreak). I haven't tested this but it may well block for a period.

**Reply by skylin008** [reply]
Thank you @peterhinch, the LIN bus break signal keep low level is 680uS .

**Reply by peterhinch** [reply]
Blocking for 680μs is unlikely to be a problem for an `asyncio` application. But the question is for how long [uart.sendbreak](https://docs.micropython.org/en/latest/library/machine.UART.html#machine.UART.sendbreak) blocks. This may be baudrate and/or platform dependent. You might like to measure it:
```python
from time import ticks_ms, ticks_diff
# Initialise UART to your requirements
t = ticks_ms()
uart.sendbreak()
dt = ticks_diff(ticks_ms(), t)
print(dt)
```

**Reply by robert-hh** [reply]
The minimum length for sendbreak is 13 bit times. So 680µs is about  13 bit times at 19200 baud.

**Reply by skylin008** [reply]
Thanks @peterhinch  @robert-hh, I had checked the send break signal via oscilloscope， it only 570uS

**Reply by skylin008** [reply]
Thanks @peterhinch  @robert-hh, I had checked the send break signal via oscilloscope display， it only 570uS

**Reply by robert-hh** [reply]
Looking at the STM32 manual, it pulls the output low for 10/11+stop bit # time, which is 11 bit, when configured with  1 stop bit. So you could temporarily reconfigure UART to 2 Stop bits + parity, send a break, and then configure back, or configure the UART temporarily to a lower baudrate, send break and them put it back.
